mlflow/metrics/__init__.py

```diff
@@ -246,41 +246,37 @@ def precision_at_k(k) -> EvaluationMetric:
+    tuple of doc ID strings.
+
+    The ``targets`` parameter should specify the column name of the ground-truth relevant doc IDs.
+
+    If you choose to use a static dataset, the ``predictions`` parameter should specify the column
+    name of the retrieved relevant doc IDs. Alternatively, if you choose to specify a function for
+    the ``model`` parameter, the function should take a Pandas DataFrame as input and return a
+    Pandas DataFrame with a column of retrieved relevant doc IDs, specified by the ``predictions``
     parameter.
 
+    ``k`` should be a positive integer specifying the number of retrieved doc IDs to consider for
+    each input query. ``k`` defaults to 3.
+
     This metric computes a score between 0 and 1 for each row representing the precision of the
-    retriever model at the given k value. The score is calculated by dividing the number of relevant
-    documents retrieved by the total number of documents retrieved or k, whichever is smaller.
-    If no relevant documents are retrieved, the score is 1, indication that no false positives were
-    retrieved.
-
-    The model output should be a pandas dataframe with a column containing a tuple of strings on
-    each row. The strings in the tuple represent the document IDs.
-    The label column should contain a tuple of strings representing the relevant
-    document IDs for each row, provided by the input ``data`` parameter.
-    The ``k`` parameter should be a positive integer representing the number of retrieved documents
-    to evaluate for each row. ``k`` defaults to 3.
-
-    This metric is a default metric for the ``retriever`` model type.
-    When the model type is ``"retriever"``, this metric will be calculated automatically with the
-    default ``k`` value of 3. To use another ``k`` value, use the ``evaluator_config`` parameter
-    in the ``mlflow.evaluate()`` API as follows: ``evaluator_config={"k": <k_value>}``.
-    Alternatively, you can directly specify the ``mlflow.metrics.precision_at_k(<k_value>)`` metric
-    in the ``extra_metrics`` parameter of the ``mlflow.evaluate()`` API without specifying a model
-    type. In this case, the ``k`` value specified in the ``evaluator_config`` parameter will be
-    ignored.
+    retriever model at the given ``k`` value. If no relevant documents are retrieved, the score is
+    0, indicating that no relevant docs were retrieved. Let ``x = min(k, # of retrieved doc IDs)``.
+    Then, the precision at k is calculated as follows:
+
+        ``precision_at_k`` = (# of relevant retrieved doc IDs in top-``x`` ranked docs) / ``x``.
+
+    This metric is a builtin metric for the ``'retriever'`` model type, meaning it will be
+    automatically calculated with a default ``k`` value of 3. To use another ``k`` value, you have
+    two options with the :py:func:`mlflow.evaluate` API:
+
+    1. ``evaluator_config={"k": 5}``
+    2. ``extra_metrics = [mlflow.metrics.precision_at_k(k=5)]``
```

Comment:
It's recommended to remove the model type in case 2. If not, I think both precision@3 and precision@5 will be calculated at the same time.